### ISSUE TITLE
Mouse Coordinates Not Detected

### ISSUE THREAD

**Issue by blisstik** [libretro/RetroArch#7697]
## Description
According to this page, event indexes should be used for using a mouse.
https://github.com/RetroPie/RetroPie-Setup/wiki/Spinners,-Trackballs,-Lightguns,-and-other-Mouse-Devices

Here is the a log of a game with the mouse attached. As you can see the Mouse #0 is attached to event2. In RetroArch Input settings, I set the `User 1 Mouse Index` to `0` and I'm able to map the mouse buttons.

The problem is I'm not able to get any coordinate movements from the mouse. There is a warning about `Full-screen pointer won't be available`. What is causing this issue?

https://pastebin.com/raw/A5e7eDLR

```
...
[INFO] [udev]: Mouse #0 (/dev/input/event2).
[INFO] [udev]: Mouse #1 (/dev/input/mouse0).
[INFO] [udev]: Plugged pad: MadCatz Madcatz Fightpad (1848:12933) on port #0.
[INFO] [udev]: Pad #0 (/dev/input/event0) supports 0 force feedback effects.
[INFO] [Autoconf]: 3 profiles found.
[INFO] [autoconf]: selected configuration: /home/pigaming/.config/retroarch/autoconfig/MadCatz Madcatz Fightpad.cfg
[INFO] [Joypad]: Found joypad driver: "udev".
[WARN] [udev]: Full-screen pointer won't be available.
...
```

### Expected behavior

Mouse coordinates to move.

### Actual behavior

Mouse coordinates are not moving but mouse buttons can be mapped without issue. I've ran a test using advmame and the coordinates move for the advmame test.

https://pastebin.com/raw/JRbHiH0T

### Steps to reproduce the bug

On an Odroid XU4, using the ORA v1.5.2 image

1. Update RetroPie-Setup script
2. Update RetroArch
3. Update lr-reicast
4. Attach a mouse
5. Launch a game like Sports Shooting USA

### Version/Commit
- retroarch - bd64deb
- lr-reicast - a45420f

### Environment information

- OS: Ubuntu 16.04.3 LTS (GNU/Linux 3.10.105-141 armv7l)

**Comment by singularity098** [comment]
Hi.  I came here to log what I believe is this same issue.  I wanted to share some more information and ask if you could confirm that information on your system to see if we can help narrow this down.

What I noticed is that I cannot map the mouse movement at all, as you have said.  I started to wonder if this was core-specific or platform specific.  Here is what I found in testing:

MAME (latest git core):  _Fail_
MAME 2016: _Fail_
MAME 2014: _Fail_
MAME 2010: **Pass**!  I can use the mouse/trackball.
Final Burn Alpha:  _Fail_
SNES9x: **Pass**!  I was able to play Mario Paint after selecting mouse in the core specific input options.

Interesting... Additionally, I tested some other environments.  This was my result:

RetroArch 1.7.5 in Ubuntu 18.10:  _Failures_ as described above
RetroArch 1.7.3 in Lakka 2.1.1:  _Failures_ as described above
RetroArch 1.7.5 in Windows 10:  **Pass**!  I did not experience any of the core-specific issues described above, the mouse works in all of the cores.

Is any of this consistent with what you're seeing?  Thank you, sir.  I hope this issue can get resolved!

**Comment by 6alileo** [comment]
The only other core I tried was Terminator 2 using lr-mame2003-plus. Oddly enough, the pointer only worked vertically.

**Comment by grant2258** [comment]
ive put a pull req in for a udev  fix if its accepted it will hopefully fix your issues what mouse are you using btw ?

**Comment by 6alileo** [comment]
I've tried a SteelSeries wired mouse and a logitech 2.4ghz mouse.

Galileo Morales
Web Developer & IT Solutions
916.601.2400

________________________________

**Comment by grant2258** [comment]
https://retropie.org.uk/forum/topic/20004/mouse-as-a-zapper-lr-fceumm-rpi-3-b/38

should get you up and going ket me know how you get on here or the forum

**Comment by grant2258** [comment]
try the fceumm with duck hunt.   see if the mouse is working there issue could be be the mouse code in the core it self if it does.  I peronally dont have any abs mouse devices.

**Comment by grant2258** [comment]
well that would make sense if the op had these core options set on windows machines  and not in the linux machines.

**Comment by 6alileo** [comment]
I tried fceumm with Duck Hunt and I was able to get it to work. I didn't have to compile RetroArch with the PR.

**Comment by grant2258** [comment]
yea well that means you mouse supports abs people that dont have this support will need that patch. It must just be the core option in your case hopefully.

**Comment by 6alileo** [comment]
Can you elaborate on mouse abs? Just a quick explanation why certain mouse won't work?

Since lr-fceumm work and lr-reicast doesn't, probably means the lightgun api isn't properly coded. It's looking like it's not a retroarch issue.

________________________________

**Comment by 6alileo** [comment]
I'm using udev and trying to get lr-reicast - Ninja Assault to work. I'm able to map the mouse buttons but coordinates will not work.

In the same environment and RA version, lr-fceumm works after turning on the zapper as mouse core option. Lr-reicast doesn't have this feature.


________________________________

**Comment by grant2258** [comment]
https://github.com/libretro/reicast-emulator/blob/37da602d6421b755ae045171d17c8d4328c7dcc0/core/libretro/libretro.cpp#L438-L445

have you looked in quick menu->controls-> user 1 device type and changing that to mouse ?

**Comment by 6alileo** [comment]
Yes. I've tried mouse and lightgun. This setting applies concurrently with the input 1 setting  as well.

________________________________

**Comment by 6alileo** [comment]
What I meant was changing input 1 to mouse also changes it in quick menu controls.

________________________________

**Comment by grant2258** [comment]
I would pot an issue on the github over there i dont have any games for this core to test it

**Comment by barbudreadmon** [comment]
On odroid, fbalpha is working as expected with games using analogs when setting mouse as control, i'll try to hook the pointer/lightgun libretro device (they are basically the same) this week-end to see if i'm experiencing issues on my odroid with it.

**Comment by barbudreadmon** [comment]
The pointer is also working as expected in fbalpha, so the issue might be with the cores indeed, a wild guess would be some type conversion going bad on arm, 2 related issues i had to deal with in the past : 
- char, which is unsigned by default (unlike x86)
- implicit conversion of negative float to unsigned int

**Comment by barbudreadmon** [comment]
Ok, i figured out the issue, i'll go into details, there are 3 ways to deal with mouse/pointer/lightgun devices in RA : 
- using `RETRO_DEVICE_MOUSE`, it's detecting mouse relative movements, basically it works like an analog stick
- using `RETRO_DEVICE_POINTER`, it's detecting mouse/touch absolute position, so you can set the cursor in the game directly to coordinates
- using `RETRO_DEVICE_LIGHTGUN`, it's detecting lightgun absolute position, so you can set the cursor in the game directly to coordinates

The deal here is that we are using `RETRO_DEVICE_LIGHTGUN` in lr-reicast (and probably in a lot of other cores), because mouse is actually able to emulate a lightgun **on a desktop computer** (might be something related to the usage of a window manager system, or some specific code in RA, idk), but on odroid (and probably some other platforms) the mouse doesn't seem to emulate the lightgun, so we'll have to implement a `RETRO_DEVICE_POINTER` device as a workaround to get this stuff to work.

**Comment by zoltanvb** [comment]
Support of mouse/pointer/lightgun has been aligned recently (#17821 and co.), so hopefully this should be in order nowadays, if not earlier.
